source/DEVS-Network/DEVSNetworkNode.class.st

```diff
@@ -0,0 +1,51 @@
+Class {
+	#name : 'DEVSNetworkNode',
+	#superclass : 'DEVSAtomicComponent',
+	#instVars : [
+		'state',
+		'messages',
+		'routingTable'
+	],
+	#category : 'DEVS-Network',
+	#package : 'DEVS-Network'
+}
+
+{ #category : 'modeling' }
+DEVSNetworkNode >> externalTransition [
+
+	| msg |
+	msg := self peekFrom: #in.
+	messages add: msg.
+	state := #processing 
+]
+
+{ #category : 'modeling' }
+DEVSNetworkNode >> initialize [ 
+	super initialize.
+	self addInputPortNamed: #in.
+	self addOutputPortNamed: #out.
+	state := #idle.
+	messages := OrderedCollection new.
+	routingTable := Dictionary new
+]
+
+{ #category : 'modeling' }
+DEVSNetworkNode >> internalTransition [
+	messages ifEmpty: [  
+		state := #idle ]
+]
+
+{ #category : 'modeling' }
+DEVSNetworkNode >> outputFunction [
+	| msg |
+	(state = #processing) ifTrue: [ 
+		msg := messages removeFirst.
+		self poke: msg to: msg to ].
```

Comment:
it should be rather `self poke: msg to: #out` since msg to is address and not id of output port.

Reply:
The address is the name of the output port. Again that is a way to make it work. Not sure if this is the right way to do it

Reply:
As stated on our talk - network node will hold own output port per each UDP server (identified by address).